pkg/capacity/pci.go

```diff
@@ -31,7 +35,7 @@ type Vendor struct {
 	// Name of the vendor
 	Name string
 	// All known devices by this vendor
-	Devices map[uint16]Device
+	Devices map[string]Device
```

Comment:
This is not a good structure right now because it does not reflect the nature hierarchy of the devices . Instead this can go as
```
Devices map[uint16]Device
```
then device itself becomes 
```
type Device struct {
   ID uint16
   Name string
   Subdevices []SubDevice
}
```
Where subdevices now can has a list of subdevices which only holds subdevice information (including sub vendor and sub device id and name

Reply:
makes sense, and the user calling `pci.GetDevice()` gets the Device with its Subdevices and I should add a `pci.GetSubdevice` that returns `(Subdevice, bool)`?

Reply:
Yes, that is possible. the Device id returned by the API should still be only `slot/vendor/device` ids without the subdevice id, but the Information can include name, and subsystem vendor and device.

Because the ID is still unique (because of the slot) so no extra information is needed.